lib/Controller/WopiController.php

```diff
@@ -196,12 +196,27 @@ public function checkFileInfo($fileId, $access_token) {
 
 		if ($this->shouldWatermark($isPublic, $wopi->getEditorUid(), $fileId, $wopi)) {
 			$email = $user !== null && !$isPublic ? $user->getEMailAddress() : "";
+			if (!empty($_SERVER['HTTP_CLIENT_IP']))   
```

Comment:
The issue with this is that the request to the checkFileInfo endpoint is not done by the user but by the Collabora server, so we will not get the actual users ip here unless it is somehow sent to us by the Collabora server. 

cc @timar @thebearon I think that would be a good enhancement to the watermarking feature. Would this be something you could imagine implementing on the Collabora side, so that the WOPI requests would send over the original IP of the user in some additional header?